src/main/java/ch/njol/skript/Skript.java

```diff
@@ -1644,178 +1644,174 @@ public static void markErrored() {
 	 * @return an EmptyStacktraceException to throw if code execution should terminate.
 	 */
-	public static EmptyStacktraceException exception(@Nullable Throwable cause, final @Nullable Thread thread, final @Nullable TriggerItem item, final String... info) {
+	public static EmptyStacktraceException exception(@Nullable Throwable cause, @Nullable Thread thread, @Nullable TriggerItem item, String... info) {
 		errored = true;
 
-		// Don't send full exception message again, when caught exception (likely) comes from this method
+		// Avoid re-throwing the same exception
 		if (cause instanceof EmptyStacktraceException) {
 			return new EmptyStacktraceException();
 		}
 
 		// First error: gather plugin package information
 		if (!checkedPlugins) {
-			for (Plugin plugin : Bukkit.getPluginManager().getPlugins()) {
-				if (plugin.getName().equals("Skript")) // Don't track myself!
-					continue;
-
-				PluginDescriptionFile desc = plugin.getDescription();
-				if (desc.getDepend().contains("Skript") || desc.getSoftDepend().contains("Skript")) {
-					// Take actual main class out from the qualified name
-					String[] parts = desc.getMain().split("\\."); // . is special in regexes...
-					StringBuilder name = new StringBuilder(desc.getMain().length());
-					for (int i = 0; i < parts.length - 1; i++) {
-						name.append(parts[i]).append('.');
-					}
+			initializePluginPackages();
+			checkedPlugins = true; // No need to do this next time
+		}
+
+		logErrorDetails(cause, info, thread, item);
+		return new EmptyStacktraceException();
+	}
+
+	private static void initializePluginPackages() {
+		for (Plugin plugin : Bukkit.getPluginManager().getPlugins()) {
+			if (plugin.getName().equals("Skript")) // Skip self
+				continue;
 
-					// Put this to map
-					pluginPackages.put(name.toString(), desc);
-					if (Skript.debug())
-						Skript.info("Identified potential addon: " + desc.getFullName() + " (" + name.toString() + ")");
+			PluginDescriptionFile desc = plugin.getDescription();
+			if (desc.getDepend().contains("Skript") || desc.getSoftDepend().contains("Skript")) {
+				String mainClassPackage = getPackageName(desc.getMain());
+				pluginPackages.put(mainClassPackage, desc);
+				if (Skript.debug()) {
+					Skript.info("Identified potential addon: " + desc.getFullName() + " (" + mainClassPackage + ")");
 				}
 			}
-
-			checkedPlugins = true; // No need to do this next time
 		}
+	}
 
+	private static String getPackageName(String qualifiedClassName) {
+		int lastDotIndex = qualifiedClassName.lastIndexOf('.');
+		return (lastDotIndex == -1) ? "" : qualifiedClassName.substring(0, lastDotIndex);
+	}
+
+	private static void logErrorDetails(@Nullable Throwable cause, String[] info, @Nullable Thread thread, @Nullable TriggerItem item) {
 		String issuesUrl = "https://github.com/SkriptLang/Skript/issues";
+		String downloadUrl = "https://github.com/SkriptLang/Skript/releases/latest";
```

Comment:
It might be better to grab this from the update checker

Though I don't think the update checker currently records this for GitHub, so a TODO comment would suffice for now

Reply:
Wouldn't using `latest` suffice for this? I don't get whats different.

Reply:
I just mean that the update checker might not always point to GitHub (e.g. in the future it could recommend Spigot if that's where you downloaded Skript from)